code/game/atoms_movable.dm

```diff
@@ -662,3 +662,212 @@
+			continue
+
+		var/datum/tilt_crit/crit_case = forced_crit
+		if(isnull(forced_crit) && should_crit)
+			crit_case = choose_crush_crit(target)
+		// note that it could still be null after this point, in which case it won't crit
+		var/damage_to_deal = crush_damage
+
+		if(isliving(target))
+			var/mob/living/L = target
+
+			if(crit_case)
+				damage_to_deal *= crit_damage_factor
+			if(iscarbon(L))
+				handle_squish_carbon(L, damage_to_deal, crit_case)
+			else
+				L.apply_damage(damage_to_deal, BRUTE)
+			L.Weaken(weaken_time)
+			L.emote("scream")
+			L.KnockDown(knockdown_time)
+			playsound(L, 'sound/effects/blobattack.ogg', 40, TRUE)
+			playsound(L, 'sound/effects/splat.ogg', 50, TRUE)
+			add_attack_logs(src, L, "crushed by [src]")
+
+
+		else if(isobj(target))  // don't crush things on the floor, that'd probably be annoying
+			var/obj/O = target
+			O.take_damage(damage_to_deal, BRUTE, "", FALSE)
+		else
+			continue
+
+		target.visible_message(
+			"<span class='danger'>[target] is crushed by [src]!</span>",
+			"<span class='userdanger'>[src] crushes you!</span>",
+			"<span class='warning'>You hear a loud crunch!</span>"
+		)
+
+	tilt_over(target_turf, angle, should_rotate, rightable, block_interactions_until_righted)
+	// for things that trigger on Crossed()
+	if(!has_tried_to_move)
+		Move(target_turf, crush_dir)
+
+	return TRUE
+
+#undef NO_CRUSH_DIR
+
+/**
+ * Tip over an atom without too much fuss. This won't cause damage to anything, and just rotates the thing and (optionally) adds the component.
+ *
+ * Arguments:
+ * * target - The turf to tilt over onto
+ * * rotation_angle - The angle to rotate by. If not given, defaults to random rotating by 90 degrees clockwise or counterclockwise
+ * * should_rotate - Whether or not we should rotate at all
+ * * rightable - Whether or not this object should be rightable, attaching the tilted component to it
+ * * block_interactions_until_righted - If true, this object will need to be righted before it can be interacted with
+ */
+/atom/movable/proc/tilt_over(turf/target, rotation_angle, should_rotate, rightable, block_interactions_until_righted)
+	visible_message("<span class='danger'>[src] tips over!</span>", "<span class='danger'>You hear a loud crash!</span>")
+	playsound(src, "sound/effects/bang.ogg", 100, TRUE)
+	var/rot_angle = rotation_angle ? rotation_angle : pick(90, -90)
+	if(should_rotate)
+		var/matrix/to_turn = turn(transform, rot_angle)
+		animate(src, transform = to_turn, 0.2 SECONDS)
+	if(target && target != get_turf(src))
+		throw_at(target, 1, 1, spin = FALSE)
+	if(rightable)
+		layer = ABOVE_MOB_LAYER
+		AddComponent(/datum/component/tilted, 14 SECONDS, block_interactions_until_righted, rot_angle)
+
+/// Untilt a tilted object.
+/atom/movable/proc/untilt(mob/living/user, duration = 10 SECONDS)
+	if(!GetComponent(/datum/component/tilted))
+		return
```

Comment:
I kinda think this proc should be baked into the component itself. Registering on the untilt signal maybe?

Reply:
Honestly, I was back and forth on this. It makes it a lot easier to until through build mode and/or proc calls to do this, tho